src/contracts/middleware/Middleware.sol

```diff
@@ -310,14 +285,31 @@ contract Middleware is SimpleKeyRegistry32, Ownable {
     function setGateway(
         address _gateway
     ) external onlyOwner {
-        gateway = IOGateway(_gateway);
+        s_gateway = IOGateway(_gateway);
     }
 
-    // function submission(bytes memory payload, bytes32[] memory signatures) public updateStakeCache(getCurrentEpoch()) {
-    //     // validate signatures
-    //     // validate payload
-    //     // process payload
-    // }
+    function setRewardsContracts(address stakerRewardsAddress, address operatorRewardsAddress) external onlyOwner {
+        s_stakerRewards = IODefaultStakerRewards(stakerRewardsAddress);
+        s_operatorRewards = IODefaultOperatorRewards(operatorRewardsAddress);
+
+        emit RewardsContractsSet(stakerRewardsAddress, operatorRewardsAddress);
+    }
+
+    function distributeRewards(
+        bytes calldata data
+    ) external onlyGateway {
+        (
+            uint256 timestamp,
```

Comment:
![image](https://github.com/user-attachments/assets/8e78e36c-1dea-4a89-a055-f4d85ca4c31a)

Yes you are missing the era index. This is the tanssi-epoch. Essentially, everything needs to be indexed by that